clientconn.go

```diff
@@ -1272,6 +1272,12 @@ func (ac *addrConn) createTransport(addr resolver.Address, copts transport.Conne
 		copts.ChannelzParentID = ac.channelzID
 	}
 
+	// Balancer implementations can specify arbitrary data in the Attributes
+	// field of resolver.Address, which is shoved into connectCtx that is passed
+	// to the transport layer. The transport layer passes the same context to
+	// the credential handshaker. This makes is possible for address specific
+	// arbitrary data from balancers to reach the credential handshaker.
+	connectCtx = credentials.WithAddressInfo(connectCtx, credentials.AddressInfo{Attr: addr.Attributes})
```

Comment:
I can't decide whether this should happen here or in the transport.  We give the transport the address already, so this is somewhat redundant here.  Credentials are partially a grpc concept but also are transport dependent.  I.e. with custom transports, an in-process implementation won't support transport credentials (it's not a byte transport so doesn't make sense).

Reply:
If we have the transport layer do it, then every transport implementation would have to do it, right. And for transports that don't support transportCredentials, this should still work (although it would be wasted effort here). But it should not affect correctness, right. Am I missing something?

Reply:
Yeah this is not about "functional correctness", but "design correctness" - making the right component responsible for doing the work.

If the API for `ClientHandshake` was to pass `resolver.Address`, which is how we'd design it now if we could, then we wouldn't do this at all.  So then the question is, which component "owns" the `ClientHandshake` API?  gRPC or Transport(s)?  Another consideration: we already pass _some_ information to the per-RPC credentials via context, and that is done in the transport.  So putting this here and that there is a little inconsistent as well.

Reply:
Moved to the transport layer. So, there are no diffs now in clientconn.go